From b90e4f68af83c92d3c7dcda3118a7c8a35f56c03 Mon Sep 17 00:00:00 2001
From: Stefan Chulski <stefanc@marvell.com>
Date: Thu, 13 Jul 2017 17:28:26 +0300
Subject: [PATCH 1077/1200] phy: net: Increase autoneg timeout from 4 to 8
 seconds

Time of link auto negation depend on link partner and could take
more than 4 seconds.
This patch increases the timeout, avoiding possible auto negotiation
timeouts.

Change-Id: Ie172893d0df6b7351d7c3218dbfa3a316d8f9dad
Signed-off-by: Stefan Chulski <stefanc@marvell.com>
Reviewed-on: http://vgitil04.il.marvell.com:8080/41610
Tested-by: iSoC Platform CI <ykjenk@marvell.com>
Reviewed-by: Igal Liberman <igall@marvell.com>
Reviewed-on: https://sj1git1.cavium.com/5832
Tested-by: sa_ip-sw-jenkins
Reviewed-by: Kostya Porotchkin <Kostya.Porotchkin@cavium.com>
---
 drivers/net/phy/phy.c | 2 +-
 include/phy.h         | 2 +-
 2 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/drivers/net/phy/phy.c b/drivers/net/phy/phy.c
index 1d8e81be2d..bcd38aafde 100644
--- a/drivers/net/phy/phy.c
+++ b/drivers/net/phy/phy.c
@@ -259,7 +259,7 @@ int genphy_update_link(struct phy_device *phydev)
 				return -EINTR;
 			}
 
-			if ((i++ % 500) == 0)
+			if ((i++ % 1000) == 0)
 				printf(".");
 
 			udelay(1000);	/* 1 ms */
diff --git a/include/phy.h b/include/phy.h
index 38469ed7f0..143de1debe 100644
--- a/include/phy.h
+++ b/include/phy.h
@@ -49,7 +49,7 @@
 				SUPPORTED_10000baseT_Full)
 
 #ifndef PHY_ANEG_TIMEOUT
-#define PHY_ANEG_TIMEOUT	4000
+#define PHY_ANEG_TIMEOUT	8000
 #endif
 
 
-- 
2.22.0

